lighthouse-core/audits/byte-efficiency/large-javascript-libraries.js

```diff
@@ -0,0 +1,167 @@
+  description: 'Large JavaScript libraries can lead to poor performance. ' +
+    'Prefer smaller, functionally equivalent libraries to reduce your bundle size.' +
+    ' [Learn more](https://developers.google.com/web/fundamentals/performance/webpack/decrease-frontend-size#optimize_dependencies).',
+  /** Label for a column in a data table. Entries will be names of large JavaScript libraries that could be replaced. */
+  name: 'Library',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class LargeJavascriptLibraries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'large-javascript-libraries',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['Stacks'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {Array<{original: BundlePhobiaLibrary, suggestions: BundlePhobiaLibrary[]}>} */
+    const libraryPairings = [];
+    const detectedLibs = artifacts.Stacks.filter(stack => stack.detector === 'js');
+
+    const seenLibraries = new Set();
+
+    for (const detectedLib of detectedLibs) {
+      if (!detectedLib.npm || !libStats[detectedLib.npm]) continue;
+      const suggestions = librarySuggestions[detectedLib.npm] || [];
+
+      if (seenLibraries.has(detectedLib.npm)) continue;
+      seenLibraries.add(detectedLib.npm);
+
+      let version = 'latest';
+      if (detectedLib.version && libStats[detectedLib.npm][detectedLib.version]) {
+        version = detectedLib.version;
+      }
+
+      const originalLib = libStats[detectedLib.npm][version];
+      let smallerSuggestions = suggestions.map(suggestion => {
+        if (!libStats[suggestion]) return;
+        if (libStats[suggestion]['latest'].gzip > originalLib.gzip) return;
+
+        return {
+          name: suggestion,
+          repository: libStats[suggestion].repository,
+          gzip: libStats[suggestion]['latest'].gzip,
+        };
+      });
+
+      smallerSuggestions = [...smallerSuggestions].sort((a, b) => a.gzip - b.gzip);
+      if (smallerSuggestions.length) {
+        libraryPairings.push({
+          original: {
+            gzip: originalLib.gzip,
+            name: detectedLib.npm,
+            repository: libStats[detectedLib.npm].repository,
+          },
+          suggestions: smallerSuggestions,
+        });
+      }
+    }
+
+    const tableDetails = [];
+    for (const libraryPairing of libraryPairings) {
+      const original = libraryPairing.original;
+      const suggestions = libraryPairing.suggestions;
+
+      tableDetails.push({
+        name: {
+          text: original.name,
+          url: original.repository,
+          type: 'link',
+        },
+        size: original.gzip,
```

Comment:
we try to be explicit now about whats the transfer and what's the resourceSize
```suggestion
        transferSize: original.gzip,
```